src/libraries/System.Diagnostics.Process/src/System/Diagnostics/PerformanceCounterLib.cs

```diff
@@ -36,7 +36,7 @@ internal PerformanceCounterLib(string machineName, string lcid)
         }
 
         /// <internalonly/>
-        internal static string ComputerName => LazyInitializer.EnsureInitialized(ref s_computerName, ref s_internalSyncObject, () => Interop.Kernel32.GetComputerName());
+        internal static string? ComputerName => LazyInitializer.EnsureInitialized<string>(ref s_computerName, ref s_internalSyncObject, () => Interop.Kernel32.GetComputerName()!);
```

Comment:
`ComputerName` doesn't seem to be nullable

Reply:
null is a possible return value for the `GetComputerName()` function. In turn this stems from the [GetComputerNameW](https://docs.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-getcomputernamew#return-value) native function.

Reply:
If GetComputerName can legitimately return null, we shouldn't be `!`'ing it.  EnsureInitialized should take a `string?` as its generic method type argument.

Reply:
And if the issue is that this overload of `EnsureInitialized` requires its T to be non-nullable, which it does if memory serves, that highlights another problem: if GetComputerName can legitimately return null and it does in some circumstance, then we're going to be calling this method every single call to ComputerName, because each time EnsureINitialized will use the field being null to suggest it hasn't yet been initialized.  In that case we'd want to use an overload that takes a bool that tracks whether it's been initialized.  Or prove to ourselves that GetComputerName can't ever return null here, or make it `GetComputerName() ?? string.Empty`, or some such thing.